src/NDItk/MultigroupTable.hpp

```diff
@@ -83,14 +114,12 @@ class MultigroupTable {
   void print( OutputIterator& iter ) const {
 
     this->metadata_.print( iter );
-    this->structure_.print( iter );
+    this->primary_structure_.print( iter );
+    for ( const auto& entry : this->outgoing_structure_ ) { entry.print( iter ); }
     this->weights_.print( iter );
     this->xs_.print( iter );
     this->release_.print( iter );
-    *iter++ = 'e';
-    *iter++ = 'n';
-    *iter++ = 'd';
-    *iter++ = '\n';
+    base::Keyword( "end" ).print( iter );
```

Comment:
No need for a "\n"?

Reply:
the print function actually adds a '\n' after the keyword.